objects/resource/ItemResource.gd

```diff
@@ -0,0 +1,63 @@
+class_name ItemResource
+extends StaticBody2D
+
+@onready var animated_sprite: AnimatedSprite2D = %AnimatedSprite
+@onready var information_panel: PanelContainer = %InformationPanel
+@onready var key_label: Label = %KeyLabel
+
+
+#region built-in
+
+func _ready() -> void:
+	information_panel.hide()
+	_display_action_key()
+
+#endregion built-in
+
+#region logic
+
+func is_collectable() -> bool:
+	return true
```

Comment:
Maybe print some error?
like `printerr("_spawn_item_data() not implemented", self)`